config/config.go

```diff
@@ -60,62 +59,64 @@ func NewDefaultConfig() *AgentConfig {
 
 // AgentConfig is the top level configuration struct
 type AgentConfig struct {
-	Admin   AdminConfig   `yaml:"admin"`
-	API     APIConfig     `yaml:"api"`
-	Log     LogConfig     `yaml:"log"`
-	Optly   OptlyConfig   `yaml:"optly"`
-	Server  ServerConfig  `yaml:"server"`
-	Webhook WebhookConfig `yaml:"webhook"`
+	Version string `yaml:"version" json:"version"`
+	Author  string `yaml:"author" json:"author"`
+	Name    string `yaml:"name" json:"name"`
+
+	SDKKeys []string `yaml:"sdkkeys" json:"sdkkeys"`
+
+	Admin     AdminConfig     `yaml:"admin" json:"admin"`
+	API       APIConfig       `yaml:"api" json:"api"`
+	Log       LogConfig       `yaml:"log" json:"log"`
+	Processor ProcessorConfig `yaml:"processor" json:"processor"`
+	Server    ServerConfig    `yaml:"server" json:"server"`
+	Webhook   WebhookConfig   `yaml:"webhook" json:"webhook"`
 }
 
 // OptlyConfig holds the set of SDK keys to bootstrap during initialization
 type OptlyConfig struct {
-	Processor ProcessorConfig `yaml:"processor"`
-	SDKKeys   []string        `yaml:"sdkkeys"`
+	Processor ProcessorConfig `yaml:"processor" json:"processor"`
 }
 
 // ProcessorConfig holds the configuration options for the Optimizely Event Processor.
 type ProcessorConfig struct {
-	BatchSize     int           `yaml:"batchSize" default:"10"`
-	QueueSize     int           `yaml:"queueSize" default:"1000"`
-	FlushInterval time.Duration `yaml:"flushInterval" default:"30s"`
+	BatchSize     int           `yaml:"batchSize" json:"batchSize" default:"10"`
```

Comment:
Nit: any lint or test we can set up to ensure both `yaml` and `json` remappings on everything?

Reply:
I can see if these two can be consolidated.

Reply:
I couldn't find one, but since `yaml` is only used for unmarshaling then we don't technically need to map the field names. We do however need the json field mappings to yield the desired casing in the `/config` endpoint so I'll remove the yaml to avoid any potential discrepancies.